### ISSUE TITLE
Knora 'Administration' docs - remarks and questions

### ISSUE THREAD

**Issue by mrivoal** [dhlab-basel/Knora#380]
Remarks
---------

- In **Permission Templates**, I can see no difference between the *OPEN* and *CLOSED* templates.

- In **Example Data stored in the permissions graph**, *Default object access permission on a ProjectMember' group*, the value of the M permission is wrong:
```
Default object access permission on a 'ProjectMember' group:
  <http://data.knora.org/permissions/[UUID]> rdf:type knora-base:DefaultObjectAccessPermission ;
       knora-base:forProject <http://data.knora.org/projects/images> ;
       knora-base:forGroup knora-base:ProjectMember ;
       knora-base:hasPermissions "CR knora-base:Creator|
                                   M <http://data.knora.org/permissions/[UUID]>|
                                   V knora-base:KnownUser"^^xsd:string .
```

- In **Example Data stored in the permissions graph**, *Administrative permissions on a 'ProjectMember' group*, `knora-base:hasPermission` is used for `knora-base:hasPermissions`
```
  <http://data.knora.org/permissions/[UUID]> rdf:type knora-base:AdministrativePermission ;
       knora-base:forProject <http://data.knora.org/projects/images> ;
       knora-base:forGroup knora-base:ProjectMember ;
       knora-base:hasPermission "ProjectResourceCreateAllPermission"^^xsd:string .
```

- I couldn't find `knora-base:hasPermissions` property neither in `knora-base.ttl` nor in `knora-admin.ttl`.

-  In *Example Project Information stored in admin named graph*, **Projects Endpoint**:
        `knora-base:projectOntolgyGraph` --> `knora-base:projectOntologyGraph`
      

Question
---------

- Given the following statement in **Administrative Permissions**

> The following permissions can be set via instances of ``knora-base:AdministrativePermission`` on any group belonging to a project. For users that are members of a number of groups with administrative permissions attached, the final set of permissions is additive and most permissive. The administrative permissions are stored in a compact format in a single string, which is the object of the predicate ``knora-base:hasPermissions`` attached to an instance of the ``knora-base:AdministrativePermission`` class. 

I am confused when I read this in **Projects Endpoint**:

> - create group named *ProjectAdmin*, give group *hasProjectAllAdminPermission* and *hasProjectResourceCreateAllPermission*
>       - create group named *ProjectMember*, give group *hasProjectResourceCreateAllPermission*,
>         *knora-base:hasDefaultChangeRightsPermission* for *knora-base:Creator*,
>         *knora-base:hasDefaultModifyPermission* for this *ProjectMember* group, and
> *knora-base:hasDefaultViewPermission* for *knora-base:KnownUser*

and the following examples from *Example Project Information stored in admin named graph*:
   ```
<http://data.knora.org/projects/[UUID]>
        rdf:type knora-base:knoraProject ;
        knora-base:projectBasepath "/imldata/SALSAH-TEST-01/images" ;
        knora-base:projectShortname "images" ;
        knora-base:projectLongname "Images Collection Demo" ;
        knora-base:projectOntolgyGraph "http://www.knora.org/ontology/images" ;
        knora-base:projectDataGraph "http://www.knora.org/data/images" ;
        knora-base:isActiveProject "true"^^xsd:boolean ;
        knora-base:hasSelfJoinEnabled "false"^^xsd:boolean .

   <http://data.knora.org/groups/[UUID]>
        rdf:type knora-base:UserGroup ;
        knora-base:groupName "ProjectAdmin" ;
        knora-base:groupDescription "Default Project Admin Group" ;
        knora-base:belongsToProject <http://data.knora.org/projects/[UUID]> ;
        knora-base:hasProjectAllAdminPermission "true"^^xsd:boolean ;
        knora-base:hasProjectResourceCreateAllPermission "true"^^xsd:boolean .

   <http://data.knora.org/groups/[UUID]>
        rdf:type knora-base:UserGroup ;
        knora-base:groupName "ProjectMember" ;
        knora-base:groupDescription "Default Project Member Group" ;
        knora-base:belongsToProject <http://data.knora.org/projects/[UUID]> ;
        knora-base:hasProjectResourceCreateAllPermission "true"^^xsd:boolean ;
        knora-base:hasDefaultChangeRightsPermission knora-base:Creator ;
        knora-base:hasDefaultViewPermission knora-base:KnownUser .
```
I would have expected
```
knora-base:hasProjectAllAdminPermission "true"^^xsd:boolean ;
knora-base:hasProjectResourceCreateAllPermission "true"^^xsd:boolean .
```
and 
```
knora-base:hasProjectResourceCreateAllPermission "true"^^xsd:boolean ;
knora-base:hasDefaultChangeRightsPermission knora-base:Creator ;
knora-base:hasDefaultViewPermission knora-base:KnownUser .
```
to be instances of `knora-base:AdministrativePermission` on groups. 

What am I missing/not understanding?

**Comment by subotic** [comment]
> In Permission Templates, I can see no difference between the OPEN and CLOSED templates.

You are right. It should be something like this:

 - Open: ``CR knora-base:Creator|M knora-base:ProjectMember|V knora-base:KnownUser``
 - Closed: ``CR knora-base:ProjectAdmin|M knora-base:ProjectMember``

Templates are not implemented at the moment.

Thanks for the corrections. Good catches :-)

> Question 1

Ups, this description is out of date. I forgot to update this part of the documentation. 

You are right. Any kind of administrative permissions need to be defined in instances of ``knora-base:AdministrativePermission`` inside the ``knora-base:hasPermissions`` string.

**Comment by subotic** [comment]
Work in progress: #409

**Comment by subotic** [comment]
Should be fixed in #409
